### PR TITLE
Add demo test screnario

### PR DESCRIPTION
**Pull request by bitbonk**
This PR is only here to demonstrate my problem described in the discussion https://github.com/GitTools/GitVersion/discussions/3177

### CODE REVIEW

**Review by asbjornu — CHANGES_REQUESTED** [review]
Thanks for providing a test! It is, however, a bit long and hard to follow. Could you please reduce it to a minimum and explain with comments which assertions in it fail and why you expect the version to be something else that what GitVersion produces?

**Comment by bitbonk** [comment]
The goal of this test was to describe the git workflow that I thought would be ideal and "suggested" as the default GitVersion flow.

The test does exactly what I described in #3177. It's the same thing but written as a test.

In this test only the last assertion fails. I think it is weird that the version on develop decrements.

I will try to create another, more reduced test.

**Comment by asbjornu** [comment]
I understand. Thank you for taking the time to write up such a thorough test. However, unless I've done something very strange, the test succeeds if you add the `fixture.Repository.MakeATaggedCommit("1.0.0");` at the very beginning, as I suggested.

File: src/GitVersion.Core.Tests/IntegrationTests/VersioningDemoScenario.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    public void ReleaseAndDevelopProblemDemo()
+    {
+        var configuration = new Config
+        {
+            // Settings the below NextVersion results in an exception
+            // I wanted to set it to globally start with version 1
+            // Setting the version to 1.0 (which is not a valid semantic version)
+            // works in GitVersion.yml but not here.
+
+            // NextVersion = "1.0"
```
[the remark this answers is not in the document]
Comment (bitbonk) [reply src/GitVersion.Core.Tests/IntegrationTests/VersioningDemoScenario.cs:22]: Making a tagged commit is a bit weird. When looking at the commit log, it looks like as if that commit was the 1.0.0 release.
That's what a version tag usually denotes.
Also I was told, that using `NextVersion` in the config is a good approach here: https://github.com/GitTools/GitVersion/discussions/3123

**Comment by bitbonk** [comment]
> I understand. Thank you for taking the time to write up such a thorough test. However, unless I've done something very strange, the test succeeds if you add the `fixture.Repository.MakeATaggedCommit("1.0.0");` at the very beginning, as I suggested.

Se my comment above

> Making a tagged commit is a bit weird. When looking at the commit log, it looks like as if that commit was the 1.0.0 release.
That's what a version tag usually denotes.
Also I was told, that using NextVersion in the config is a good approach here: https://github.com/GitTools/GitVersion/discussions/3123

**Comment by bitbonk** [comment]
@asbjornu I have created a reduced, simplified test in `ReducedReleaseWorkflowDemo`.
All red emojis mark some sort of problem? 😉

**Comment by bitbonk** [comment]
@asbjornu OK I think this whole thing drills down to:
How can make `develop` start with `1.0.0-alpha.X` in a brand new repo and at the same time have only X incremented on `develop` until the final release of `1.0.0` has actually happened (is merged into `main`)?

**Comment by asbjornu** [comment]
It is absolutely the case that `next-version` **should** give you the behavior you expect. Unfortunately, there is something in the handling of `next-version` that makes it not work like a `git tag`, which is unfortunate. For some use cases `next-version` works just fine, but for others, it does not. I consider that a bug and would like to see this fixed by having `next-version: 1.0.0` work exactly like `git tag 1.0.0`. However, until that bug is fixed, I think you will have to use `git tag` to achieve the behavior you want.

**Comment by bitbonk** [comment]
> I think you will have to use `git tag` to achieve the behavior you want.

Unfortunately tagging develop with `1.0.0` does not give me behavior I want. This will cause versions on `develop` have a `1.0.0` version (without the `-alpha`). The initial (complicated) test was just asserting that the semantic version increments that why it accidentally passed, my test was not exact enough.

**Comment by bitbonk** [comment]
I would suggest do continue discussion on #3177 instead of in this PR, if necessary.

**Comment by asbjornu** [comment]
> Unfortunately tagging develop with `1.0.0` does not give me behavior I want. This will cause versions on `develop` have a `1.0.0` version (without the `-alpha`). 

That only happens if you tag on the `develop` branch, no? Stable tags should preferably go on the `main` branch. At least, that's what GitVersion expects.

> The initial (complicated) test was just asserting that the semantic version increments that why it accidentally passed, my test was not exact enough.

I see.

**Comment by asbjornu** [comment]
Did #3190 resolve this?

**Comment by bitbonk** [comment]
> Did #3190 resolve this?

@asbjornu I was just trying to build the main branch locally, and I am getting the following two errors:

> GitVersion.Core: Invalid option '11' for /langversion. Use '/langversion:?' to list supported values.
> GitTools.Testing: Invalid option '11' for /langversion. Use '/langversion:?' to list supported values.

What are the prerequisites to get the main branch built?

**Comment by arturcic** [comment]
@bitbonk we recently merged the .net 7 support so you should install the rc2 for now, and in couple of weeks the final release of .net 7.0

**Comment by bitbonk** [comment]
OK, running the test with the new changes yield some slightly changed behavior on how the versions (_major_, _minor_, _patch_ and `-alpha.commits`) are incremented which is probably OK (as far as I can tell) but my main problem that there were scenarios where the semver on  the `develop` could actually _decrement_ does not exist anymore.

What release will include these changes?

**Comment by arturcic** [comment]
@bitbonk the new major version - v6, but we don't have an exact time we release it, but most probably soon we'll start publishing first alphas or betas of v6 to get first feedback
